l2geth/rollup/sync_service.go

```diff
@@ -138,17 +138,18 @@ func NewSyncService(ctx context.Context, cfg Config, txpool *core.TxPool, bc *co
 		}
 
 		// Wait until the remote service is done syncing
-		for {
+		t := time.NewTicker(10 * time.Second)
+		for ; true; <-t.C {
```

Comment:
Is there a possibility that the ticker goes off while the system is in the middle of syncing? Do we get two threads simultaneously trying to sync?

Reply:
> Is there a possibility that the ticker goes off while the system is in the middle of syncing?

If the processing that occurs after the tick takes longer than the tick time, then the next tick will be dropped. If the processing takes less time then the code will block on `<-t.C` until the next tick, see: https://golang.org/src/time/tick.go

```
	// Give the channel a 1-element time buffer.
	// If the client falls behind while reading, we drop ticks
	// on the floor until the client catches up.
```

> Do we get two threads simultaneously trying to sync?  If so, is there a potential problem with both threads trying to sync at the same time?

No, this code is not creating new goroutines and is not used concurrently

Reply:
The purpose of the `for ; true; <-.t.C` syntax over `for range t.C` is that with the second syntax it will not run the first iteration of the loop until after the first tick. We currently run with a 10 second or so tick which is too long to wait for on boot

Reply:
Thanks for the context. LGTM.